tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -36,6 +44,10 @@ spec:
       description: Also push a tag with the current timestamp for each image in the Snapshot
       type: string
       default: "false"
+  results:
+    - name: commonTag
+      type: string
+      description: Common tag for downstream tasks. Only set if tagPrefix length is nonzero
```

Comment:
When you say "you cannot", what do you mean exactly? What happens if you don't? I would expect it to be "" unless you save something else. Am I wrong?

Reply:
I am pretty sure tekton fails the task. I did this a little over a month ago, so I may be remembering something wrong, but I was quite sure the task errors if you have a result that is only set sometimes. I can test it after lunch and report back

Reply:
Changed. It now sets the result always.

Reply:
Just a follow up. It might fail if a subsequent task needs the result and it was not set (or the result file was not touched). It sounds like a bug because if you defined a result in the task, it should exist even if it has no content. I changed so far, but I can change it back after Johnny report back if it fails or not.

Reply:
Of course if it fails, it's an issue. But it seems the existing task tests that we have didn't start failing because of that, right? In they didn't use tagPrefix, so the result was not set. So doesn't that prove that it's ok not to set the result?

Reply:
Okay, yeah, I had tested with pipeline results, not task results. Here is an example
```
---
apiVersion: tekton.dev/v1
kind: Task
metadata:
  name: results-task
spec:
  results:
    - name: test
  steps:
    - name: echo
      image: registry.access.redhat.com/ubi8/ubi:latest
      script: |
         #!/usr/bin/env bash
         echo abc
---
apiVersion: tekton.dev/v1
kind: Pipeline
metadata:
  name: results-pipeline
spec:
  results:
    - name: testOutput
      value: $(tasks.results-task.results.test)
  tasks:
    - name: results-task
      taskRef:
        name: results-task
```
in this example, the taskrun succeeds but the pipelinerun fails. So as long as no pipeline result is set based on the task result in an instance where the task result wasn't set, I suppose you are fine

Reply:
Okay, I tried this
```
---
apiVersion: tekton.dev/v1
kind: Task
metadata:
  name: results-task
spec:
  results:
    - name: test
  steps:
    - name: echo
      image: redhat.com/ubi8/ubi:latest
      script: |
         #!/usr/bin/env bash
         echo abc
---
apiVersion: tekton.dev/v1
kind: Task
metadata:
  name: results-task2
spec:
  params:
    - name: param
      type: string
  steps:
    - name: echo
      image: redhat.com/ubi8/ubi:latest
      script: |
         #!/usr/bin/env bash
         echo abc
---
apiVersion: tekton.dev/v1
kind: Pipeline
metadata:
  name: results-pipeline
spec:
  tasks:
    - name: results-task
      taskRef:
        name: results-task
    - name: results-task2
      taskRef:
        name: results-task2
      params:
        - name: param
          value: $(tasks.results-task.results.test)
```
the pipelinerun fails, task 2 never starts. It says `Could not find result with name test for task results-task`

Reply:
Thanks. So that means my theory could be right I guess.